### ISSUE TITLE
Why bn_feat rather than feat is used in evaluation when feature before or after BN is used in training?

### ISSUE THREAD

**Issue by QQQYang** [JDAI-CV/fast-reid#231]
https://github.com/JDAI-CV/fast-reid/blob/a5550d77254faca06be4b4794c5ed6dcfa7da0bc/fastreid/modeling/heads/reduction_head.py#L55
https://github.com/JDAI-CV/fast-reid/blob/a5550d77254faca06be4b4794c5ed6dcfa7da0bc/fastreid/modeling/heads/reduction_head.py#L63

Hello, I don't know much about ReID. Can anyone help me figure out why feature adopted in training and evaluation is different?

**Comment by L1aoXingyu** [comment]
Experimental results show that performance is the best in this way.

**Comment by QQQYang** [comment]
> Experimental results show that performance is the best in this way.

Thanks for your reply. If I want to distill feature from teacher, I should use the feature before BN. Right?

**Comment by L1aoXingyu** [comment]
You can try both features before BN and after BN to see which one can get a good result.

Actually, we try to distill model with feature between teacher and student model, but cannot get a good result. Our best result can be got by kl loss from classification logits between teacher and student model.

**Comment by QQQYang** [comment]
@L1aoXingyu Great thanks for sharing your experience. I will try KL loss next. Yesterday, I distilled features before BN by directly using L2 loss, it seems to work fine.
